File: client/src/insight_charts/caption/fact-templates-zh/univariate_distribution.js
const templates = [
    {
        'id': 0,
        'template': '{{action}}在{{subspace0}}到{{subspace1}} 之间，{{agg}}{{measure}}的数据分布如图所示. 其数据范围位于{{min}}到{{max}}区间内.',
    },
    {
        'id': 1,
        'template': '{{action}}在{{subspace0}}到{{subspace1}} 之间，{{agg}}{{measure}}的数据分布的结果如图所示. 其平均值为{{mean}}.',
    },
    {
        'id': 2,
        'template': '{{action}}在{{subspace0}}到{{subspace1}} 之间，{{agg}}{{measure}}随时间的分布如图所示. 其中最大值为{{max}}，最小值为{{min}}.',
    },
]

export default templates;

File: server/services/generator/server/app/insight/univariate_distribution.py
import pandas as pd
from typing import List

from ..common.utils import ChartTemplateConfiguration


def extract(data: pd.DataFrame, config: ChartTemplateConfiguration, **kwargs) -> List:
    """
    Find univariate_distribution insight in univariate temporal data
    :param data:
        DataFrame,Raw time series data
    :param json_requires:
        dict,The JSON content needed to draw the diagram
    :param kwargs:
        Optional parameters
    :return:
        dict,The corresponding JSON format of the visual chart
    """
    data = data[config.insight.measure[0]][config.insight.subspace[0]:config.insight.subspace[1]]
    value_field = config.insight.measure[0]
    insight = {
        'mean': float(data.mean()),
        'mean+2std': float(data.mean() + 2 * data.std()),
        'mean-2std': float(data.mean() - 2 * data.std()),
        'min': float(data.min()),
        'max': float(data.max())
    }
    return [[{'field': value_field, 'value': value, 'name': key} for key, value in insight.items()]]
